src/components/head-metadata/index.tsx

```diff
@@ -0,0 +1,52 @@
+import HashiHead from '@hashicorp/react-head'
+import { useCurrentProduct } from 'contexts'
+import { HeadMetadataProps } from './types'
+
+/**
+ * Builds up the the necessary meta tags for the site. Rendered in `_app`, where it receives `pageProps.metadata` as props
+ *
+ * We build up a page title that looks like {props.title} | {currentProduct} | {root title}
+ */
+export default function HeadMetadata(props: HeadMetadataProps) {
+  const { name: productName, slug: productSlug } = useCurrentProduct() ?? {}
+
+  const titleParts = [__config.dev_dot.meta.title]
+  const description = props.description ?? __config.dev_dot.meta.description
+
+  // We're using .unshift() to add the following elements to the beginning of the array
+  if (productName) {
+    titleParts.unshift(productName)
+  }
+
+  if (props.title) {
+    titleParts.unshift(props.title)
+  }
+
+  const title = titleParts.join(' | ')
+
+  const finalDescription = description.replace(
+    '{product}',
+    productName ?? 'HashiCorp'
+  )
+
+  const ogImageUrl = `/og-image/${productSlug ?? 'base'}.jpg`
+
+  return (
+    // TODO: OpenGraph image to be passed as the image prop here
+    <HashiHead
+      title={title}
+      siteName={title}
+      pageName={title}
+      description={finalDescription}
+      image={ogImageUrl}
+    >
+      <link rel="icon" href="/favicon.ico" sizes="any" />
+      <link rel="icon" href="/icon.svg" type="image/svg+xml" />
+      <meta name="viewport" content="width=device-width, initial-scale=1.0" />
+      <meta
+        name="google-site-verification"
+        content="zRQZqfAsOX-ypXfU0mzAIzb5rUvj5fA4Zw2jWJRN-JI"
+      />
```

Comment:
I believe that the `HashiHead`'s `image` prop is insufficient alone to render images properly on all social networks.  Let me verify this real quick...

Reply:
Yep won't render on Twitter correctly.. Lemme find what you gotta add here

<img width="1038" alt="CleanShot 2022-06-09 at 11 45 44@2x" src="https://user-images.githubusercontent.com/2105067/172921564-8ef3e5bc-abcc-48db-9b4b-ab38f692503f.png">

Reply:
I think you are right, we've been seeing some issues with twitter share cards and io sites as a result of this. Lemme check the validator.

Reply:
Good catch @BrandonRomano I think we also need `twitter:image`: https://developer.twitter.com/en/docs/twitter-for-websites/cards/overview/summary-card-with-large-image

Reply:
Here's some usage that I think illustrates all the other metadata props we have to add here for it to work on all networks:

https://github.com/hashicorp/how-hashicorp-works/blob/b8c2b775b88615d875e1d2adf92f10e829aecea2/layouts/base/index.jsx#L27-L50